ceph.spec.in

```diff
@@ -395,7 +395,6 @@ Requires:	python%{_python_buildid}-cephfs = %{_epoch_prefix}%{version}-%{release
 Requires:	python%{_python_buildid}-rgw = %{_epoch_prefix}%{version}-%{release}
 Requires:	python%{_python_buildid}-ceph-argparse = %{_epoch_prefix}%{version}-%{release}
 Requires:	python%{_python_buildid}-ceph-common = %{_epoch_prefix}%{version}-%{release}
-Requires:	python%{_python_buildid}-requests
```

Comment:
```
$ git grep -w requests|grep import|grep \.py|grep -v -w qa
src/pybind/mgr/ansible/ansible_runner_svc.py:import requests
src/pybind/mgr/ansible/module.py:import requests
src/pybind/mgr/ansible/tests/test_client_playbooks.py:from requests.exceptions import ConnectionError
src/pybind/mgr/dashboard/awsauth.py:from requests.auth import AuthBase
src/pybind/mgr/dashboard/controllers/prometheus.py:import requests
src/pybind/mgr/dashboard/grafana.py:import requests
src/pybind/mgr/dashboard/rest_client.py:import requests
src/pybind/mgr/dashboard/rest_client.py:from requests.exceptions import ConnectionError, InvalidURL, Timeout
src/pybind/mgr/dashboard/rest_client.py:    from requests.packages.urllib3.exceptions import SSLError
src/pybind/mgr/dashboard/services/iscsi_client.py:from requests.auth import HTTPBasicAuth
src/pybind/mgr/dashboard/tests/test_rest_client.py:import requests.exceptions
src/pybind/mgr/deepsea/module.py:import requests
src/pybind/mgr/influx/module.py:    from requests.exceptions import RequestException
src/pybind/mgr/telemetry/module.py:import requests
src/script/ptl-tool.py:import requests
src/test/rgw/rgw_multi/zone_cloud.py:import requests.compat
src/test/rgw/rgw_multi/zone_es.py:import requests.compat
```

so, could you add this dependency to ceph-mgr?

Reply:
`ceph-mgr` depends on `ceph-base` (https://github.com/ceph/ceph/blob/master/ceph.spec.in#L445),  and `python-requests` is a dependency of `ceph-base` (https://github.com/ceph/ceph/blob/master/ceph.spec.in#L366)

should i add the dependency to `ceph-mgr` explicitly?

Reply:
@runsisi if i read your commit message right, you intended to remove python-requests from the runtime dependencies of ceph-base. so if you want to do this, ceph-mgr won't depend on python-requests anymore after your change. my suggestion is based on your change.

Reply:
i was trying to remove python-requests  from dependencies of `ceph-common` not `ceph-base` :)

Reply:
sorry, i was wrong.

> and `python-requests` is a dependency of `ceph-base` (https://github.com/ceph/ceph/blob/master/ceph.spec.in#L366)

could you help check if this applies to debian packaging?

Reply:
we want to minimize the third party dependencies on ceph client sides to ease the setup procedure for our customers, since `/bin/ceph` (included in `ceph-common`) is needed for client side, so we have to minimize the dependencies of `ceph-common`.

some other changes i am trying to do is moving `ceph-dencoder`, `cephfs-*` and `radosgw-admin` from `ceph-common` to `ceph-base`, i think these tools are more server side related, @tchaikov any sugguestions ? thanks.

Reply:
> could you help check if this applies to debian packaging?

sorry, my fault, i forgot to check the debian changes. i should be more careful.

updated and repushed.

Reply:
actually, if you could help update it. that'd be great.

Reply:
ok, i will update alpine rules in a separate pr.